internal/controller/drplacementcontrol_controller.go

```diff
@@ -1285,6 +1268,42 @@ func (r *DRPlacementControlReconciler) finalizeDRPC(ctx context.Context, drpc *r
 	return nil
 }
 
+func (r *DRPlacementControlReconciler) cleanupForVolSyncVRGMCV(
+	ctx context.Context,
+	drPolicy *rmn.DRPolicy,
+	log logr.Logger,
+	mwu rmnutil.MWUtil,
+	drpc *rmn.DRPlacementControl,
+	vrgNamespace string,
+) error {
+	// Cleanup volsync secret-related resources (policy/plrule/binding)
+	if err := volsync.CleanupSecretPropagation(ctx, r.Client, drpc, r.Log); err != nil {
+		return fmt.Errorf("failed to clean up volsync secret-related resources (%w)", err)
+	}
+
+	drClusters, err := GetDRClusters(ctx, r.Client, drPolicy)
+	if err != nil {
+		return fmt.Errorf("failed to get drclusters. Error (%w)", err)
+	}
+
+	// Verify VRGs have been deleted
+	vrgs, _, _, err := getVRGsFromManagedClusters(r.MCVGetter, drpc, drClusters, vrgNamespace, log)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve VRGs. We'll retry later. Error (%w)", err)
+	}
+
+	if !ensureVRGsManagedByDRPC(r.Log, mwu, vrgs, drpc, vrgNamespace) {
+		return fmt.Errorf("VRG adoption in progress")
+	}
+
+	// delete MCVs
+	if err := r.deleteAllManagedClusterViews(drpc, rmnutil.DRPolicyClusterNames(drPolicy)); err != nil {
```

Comment:
We need to delete VRG MW and make sure the MCV does not return any VRG, and then delete the MCV.

The older order of functions was:
- Get VRGs (from MCV)
- Delete all VRG MW
  - Wait for VRG count to zero as per MCV
- Delete MCV
- Delete metrics

In the current scheme we delete the MCV too early and the VRG MW later, this should not be the case.